addons/html_editor/static/src/editor/link/link.js

```diff
@@ -0,0 +1,38 @@
+// import { _t } from "@web/core/l10n/translation";
+import { Component, useRef, useState, onMounted } from "@odoo/owl";
+import { useOverlay } from "../core/overlay_plugin";
+
+export class LinkPopover extends Component {
+    static template = "html_editor.linkPopover";
+    static props = {
+        // overlay props
+        dispatch: Function,
+        offset: Function,
+        el: {
+            validate: (el) => el.nodeType === Node.ELEMENT_NODE,
+        },
+        // link props
+        linkState: Object,
+    };
+    setup() {
+        console.log(this.props.linkState.linkElement);
+        // const ref = useRef("root");
+        this.urlInput = useRef("urlInput");
+        this.state = useState({ editing: true, url: "" });
+        this.overlay = useOverlay("root", { position: "bottom" });
+        onMounted(() => {
+            this.state.url = this.props.linkState.linkElement.href || "";
```

Comment:
why do you need to do it in onmounted? can't you just initialize state directly with these values?

Reply:
Ahhh, yes indeed, and it fixes the glitching when clicking on the link. Thanks!